src/main/java/me/lokka30/phantomworlds/commands/phantomworlds/subcommands/TeleportSubcommand.java

```diff
@@ -108,17 +94,77 @@ public void parseCommand(@NotNull PhantomWorlds main, CommandSender sender, Comm
      */
     @Override
     public List<String> parseTabCompletion(PhantomWorlds main, CommandSender sender, Command cmd, String label, String[] args) {
-        if (!sender.hasPermission("phantomworlds.command.phantomworlds.teleport")) {
+    	String subCommand = args[0].toLowerCase(Locale.ROOT);
+    	
+    	if (!sender.hasPermission("phantomworlds.command.phantomworlds." + subCommand)) {
             return new ArrayList<>();
```

Comment:
Note to self: I'll need to crease use of the empty lists, fellow developer of my plugin LevelledMobs found out that there's a better way of doing this.